### ISSUE TITLE
Libation v9

### ISSUE THREAD

**Issue by rmcrackan** [rmcrackan/Libation#418]
@Mbucari , @pixil98 -- To my knowledge, all 3 of us have completed all major pending efforts. I'm ready to publicly announce Libation v9 and take mac and linux out of beta. Well, maybe I'll do 1 more release first so if the latest changes broke something, they won't first come to light during a major announcement.

Anything I should wait on?

pixel98: what's your reddit name? Barring objections: when advertising it there, I want to credit you by username .

**Comment by Mbucari** [comment]
@rmcrackan Nothing more from me in the near future. I'm feeling really good about the recent changes and chardonnay refactoring.

**Comment by rmcrackan** [comment]
Good reminder: that's the other thing. We likely don't want to release v9 until the new year -- just as true today as it was [5 years ago](https://i.redd.it/8uvcmqtpv8501.png)

**Comment by pixil98** [comment]
No objections to being credited, my username is on Reddit is `pixil`. Good call on waiting until the new year. It might be worth cutting a tag now just to make sure the pipelines work correctly and it publishes to docker hub. I spent all my vacation already so I'll be around if it needs any fixes before the big release.

**Comment by wtanksleyjr** [comment]
Once you formally publish it to docker-hub, I'd love to beta-test it, I've
got an Ubuntu server and a Windows machine running Libation. I'd like to
try the obvious use case: moving an existing Windows install onto a Docker
one. I have a little bit of knowledge so I'd be happy to mess with a bit of
scripting, and definitely won't have a problem patching together a
docker-compose and editing my configs from the clues above (although
honestly I fear it might take a while to figure out, it's been discussed in
way too many threads).

-Wm

**Comment by rmcrackan** [comment]
I've initiated v8.7. I expect it live within the hour

**Comment by rmcrackan** [comment]
@pixil98

* I pushed the new v8.7.0 tag
* Some [actions](https://github.com/rmcrackan/Libation/actions/runs/3742862065) occurred successfully incl. building artifacts
* The [releases](https://github.com/rmcrackan/Libation/releases) page does not have a new draft
* My docker repo is empty

**Comment by pixil98** [comment]
For some reason the workflow didn't trigger on the tag being created. I just tried testing it on my identical forked repo and it worked fine. How are you creating and pushing the tag?

**Comment by rmcrackan** [comment]
```
git tag v8.7.0
git push --tags
```

**Comment by Mbucari** [comment]
I just tested these workflows on my repo. I created a new release with a new tag at the same time, and all of the workflows triggered as expected.

https://github.com/Mbucari/Libation/actions/runs/3745722351

**Comment by rmcrackan** [comment]
What was your method for creating a new release and new tag? I had pushed my code. Then separately created and pushed the tag

**Comment by Mbucari** [comment]
I just used GitHub's web GUI. Draft a new release > choose a tag.

Then I add a new tag, and GitHub creates it when I publish the release.

**Comment by pixil98** [comment]
Can you try deleting the tag and recreating it or creating a new tag? I don’t see a problem with how you created the tag. I generally create the tag and push it specifically, git push origin v8.7.0

**Comment by rmcrackan** [comment]
Weird. `git push origin v8.7.0` worked. This is how people become superstitious!

**Comment by rmcrackan** [comment]
Ok you docker weirdos :) it's showing up there now too

rmcrackan/libation

latest (and only) is 8.7.0

**Comment by rmcrackan** [comment]
@pixil98 , @Mbucari I don't remember where one of you said "When you announce v9.0 on reddit, can you add a request for help creating a .deb linux package?"

It looks like a script will be available even sooner -- https://github.com/rmcrackan/Libation/issues/390#issuecomment-1367452436

**Comment by rmcrackan** [comment]
All known bugs are patched. I believe these are the last items before a v9 release:

@pixil98 

* Can you work your devops magic with [this script](https://github.com/rmcrackan/Libation/blob/master/Source/targz2deb.sh) and make life easier for our Linux friends?
* All unit tests should be fixed for linux. Am I correct that they are included in the CI pipeline? And if not, is it difficult to add?

@Mbucari 

* #324 isn't a blocker so I'm fine publishing without it. There's no rush on it
* You have a to-do list in #339 which looks stale. Is anything still needed from there?

**Comment by pixil98** [comment]
I can take a look at the script this weekend. It should be doable. All of the unit tests are running, on windows and linux, for each push to main and each pull request to main.

**Comment by rmcrackan** [comment]
@pixil98 Thanks. I was hoping to release v9 by now but my family's health imploded. Publishing is easy. If anything goes awry though, I have neither the time nor mental capacity to debug it. Hopefully normalcy will return soon.

**Comment by CharlieRussel** [comment]
Seriously @rmcrackan, family is way more important than a software program. Take care and we’ll wait.

**Comment by Mbucari** [comment]
@rmcrackan I'm sorry about your family situation, but I am glad v9 got delayed. It's nice to have something new and shiny to go along with a major version release, and I think the user ratings fit that bill nicely.

Hope everyone gets better soon!

**Comment by wtanksleyjr** [comment]
I don't know anything about how to fix this, but the Docker container is
running as root, so everything it creates will be owned by root. Most
containers give some way to set the user, but I don't know how that's to be
implemented.

-Wm

**Comment by pixil98** [comment]
@wtanksleyjr oops, that not great. Can you submit an issue and I'll take care of it next chance I get?

**Comment by wtanksleyjr** [comment]
Actually, just found that was due to a mistake I'd made -- I didn't hand-create the folder, so Docker made it "for" me but with wrong permissions. Even with that fixed, though, it still won't run ... I'll submit a report, but it's almost certainly just me doing something wrong.

**Comment by rmcrackan** [comment]
@pixil98 @Mbucari I'm recovered from covid, back in town, and ready to release. Does anything need to be updated in the documentation first? Esp regarding linux or docker?

**Comment by Mbucari** [comment]
@rmcrackan I updated the linux install instructions.

**Comment by wtanksleyjr** [comment]
Where's the Docker documentation? I didn't see any before, did I miss it?
(As you know, I gave up in favor of the DEB, which creates all of the
needed files.)

-Wm

**Comment by Mbucari** [comment]
@wtanksleyjr There isn't any Docker documentation. That was all done by @pixil98, and as far as I can tell, it uses default scripts from [docker](https://github.com/docker/build-push-action).

[This is the action](https://github.com/rmcrackan/Libation/blob/master/.github/workflows/docker.yml) that guilds the docker image.

**Comment by wtanksleyjr** [comment]
Well, it's up to someone who knows Docker to decide whether it's ready to
release; I apologize but I wasn't able to make it work at all, endless
exceptions. I don't know whether anyone's reported making it work.

-Wm

**Comment by rmcrackan** [comment]
v9 is announced. Let the bug reports begin lol

**Comment by wtanksleyjr** [comment]
(Just a note that the DEB works for me, thank you!)

-Wm

**Comment by rmcrackan** [comment]
@wtanksleyjr w00h00 -- thanks for that confirmation

**Comment by jhofker** [comment]
Working great on Docker here. Most difficult part was running it initially somewhere else to get the config and DB spun up. Thanks folks!

**Comment by wtanksleyjr** [comment]
How did you run it -- what folders need to be created, what's the command
line/docker-config yaml, etc? If you don't mind? I spent hours smashing my
head into a brick wall before I gave up with no docs, no clues, zero
visible progress, same error in the logs every time, and no hints that I
was "getting warmer" or "getting colder".

-Wm

**Comment by pixil98** [comment]
I'll write up a readme for it. It should be as simple as invoking it with:
```
sudo docker run -d \
  -v <path to your existing config directroy>:/config \
  --name necesse \
  --restart=always \
  rmcrackan/libation
```

Disclaimer: I haven't actually tried running the above command.

**Comment by wtanksleyjr** [comment]
Oof. Well, I must be just unlucky. The error was something about not being
able to access the appsettings file, which of course is inside the Docker
so config shouldn't matter at all. I'll fiddle with it after I see the docs
put up.

-Wm

**Comment by jhofker** [comment]
I use docker compose, but it should be the same as just doing a `docker run`

```  
libation:
    image: rmcrackan/libation
    container_name: libation
    volumes:
      - /local/path/for/libation/config:/config
      - /local/path/for/audio/books:/data
      - /tmp/libation:/tmp
    restart: unless-stopped
```
`docker compose up -d`

Before doing that, I ran Libation on my Windows machine to set up my account. Then copied over `AccountSettings.json`, `LibationContext.db`, and `Settings.json` to `/local/path/for/libation/config` (any path on the host machine where you store config-like stuff for your running containers). I also went through `Settings.json` and made the following adjustments:

```
  "Books": "/data",
  "InProgress": "/tmp",
```
I had also updated Serilog.WriteTo.Args.path to `/var/logs/libation.log` or something like that, but it appears that change was overwritten so it may not matter.

**Comment by rmcrackan** [comment]
Thanks all. I like where this docker conversation is going. Looking forward to a readme by one of you folks who clearly know docker better than me.

Docker isn't a high-demand feature. For those power users who like it though, it'll be a really valuable addition. Much like having created the CLI

**Comment by pixil98** [comment]
@wtanksleyjr I've added a docker.md file to the Documentation folder. I tested it out and was able to configure the latest docker image from scratch using those instructions. If you can reproduce the error you were seeing following those instructions, please write up a new issue for it and we can take a look at it.
